templates/flavors/flatcar/machine-deployment.yaml

```diff
@@ -0,0 +1,85 @@
+---
+apiVersion: cluster.x-k8s.io/v1alpha4
+kind: MachineDeployment
+metadata:
+  name: "${CLUSTER_NAME}-md-0"
+spec:
+  clusterName: "${CLUSTER_NAME}"
+  replicas: ${WORKER_MACHINE_COUNT}
+  selector:
+    matchLabels:
+  template:
+    spec:
+      clusterName: "${CLUSTER_NAME}"
+      version: "${KUBERNETES_VERSION}"
+      bootstrap:
+        configRef:
+          name: "${CLUSTER_NAME}-md-0"
+          apiVersion: bootstrap.cluster.x-k8s.io/v1alpha4
+          kind: KubeadmConfigTemplate
+      infrastructureRef:
+        name: "${CLUSTER_NAME}-md-0"
+        apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+        kind: AzureMachineTemplate
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+kind: AzureMachineTemplate
+metadata:
+  name: "${CLUSTER_NAME}-md-0"
+spec:
+  template:
+    spec:
+      vmSize: ${AZURE_NODE_MACHINE_TYPE}
+      osDisk:
+        osType: "Linux"
+        diskSizeGB: 128
+      sshPublicKey: ${AZURE_SSH_PUBLIC_KEY_B64:=""}
+      image:
+        sharedGallery:
+          subscriptionID: ${AZURE_SUBSCRIPTION_ID}
+          resourceGroup: "cluster-api-images"
+          gallery: "ClusterAPI"
+          name: "capi-flatcar-stable-2905.2.3"
+          version: "0.3.1632385693"
+          publisher: "kinvolk"
+          offer: "flatcar-container-linux-free"
+          sku: "stable"
```

Comment:
It seems those values (quotes) and order of keys could be consistent with other files.

Reply:
I tried to order these field in (what I considered) logical order:
* subscription -> resource group -> gallery -> (image) name
* plan publisher -> offer ->sku
but the envsubst step seems to reorder the fields alphabetically. So i'll reorder this.

Are you suggesting removing the quotes?

Reply:
Between the files, I see:
- subscriptionID, resourceGroup, gallery, name, ...
- subscriptionID, resourceGroup, name, gallery, ...
- gallery, name, resourceGroup, sku, subscriptionID, ...

>but the envsubst step seems to reorder the fields alphabetically. So i'll reorder this.

I remember I was fighting with this as well I think it's a good idea to have them ordered as after the rendering, as then it's possible to diff between the template and actual configs (which in turn might be useful for updating your configuration).

Reply:
The fields are now in the same order as in the rendered template and quotes are gone wherever the rendered template removes them.

Reply:
Awesome, thank you!